types/share_splitting.go

```diff
@@ -51,63 +102,110 @@ func splitMessage(rawData []byte, nid namespace.ID) NamespacedShares {
 	return shares
 }
 
-// splitContiguous splits multiple raw data contiguously as shares.
-// Used for transactions, intermediate state roots, and evidence.
-func splitContiguous(nid namespace.ID, rawDatas [][]byte) []NamespacedShare {
-	shares := make([]NamespacedShare, 0)
-	// Index into the outer slice of rawDatas
-	outerIndex := 0
-	// Index into the inner slice of rawDatas
-	innerIndex := 0
-	for outerIndex < len(rawDatas) {
-		var rawData []byte
-		startIndex := 0
-		rawData, outerIndex, innerIndex, startIndex = getNextChunk(rawDatas, outerIndex, innerIndex, consts.TxShareSize)
-		rawShare := append(append(append(
-			make([]byte, 0, len(nid)+1+len(rawData)),
-			nid...),
-			byte(startIndex)),
-			rawData...)
-		paddedShare := zeroPadIfNecessary(rawShare, consts.ShareSize)
-		share := NamespacedShare{paddedShare, nid}
-		shares = append(shares, share)
-	}
-	return shares
+type ContiguousShareWriter struct {
+	shares       []NamespacedShare
+	pendingShare NamespacedShare
+	namespace    namespace.ID
+}
+
+func NewContiguousShareWriter(ns namespace.ID) *ContiguousShareWriter {
```

Comment:
Suggestion: Add a doc comment?

Reply:
not sure, but I likely accidently switched this when trying to help Tomasz debug and forgot to change it back. Changed it back here. https://github.com/celestiaorg/celestia-core/pull/637/commits/7d2505d9c2862ecbac45c4e8c93dab9de4ac7380